src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java

```diff
@@ -45,6 +52,10 @@
      * Contains all resources needed for a certain build.
      */
     private HashMap<String, ItemStack> neededResources = new HashMap<>();
+    /**
+     * Contains all resources available in the chest or builder inventory needed for a certain build.
+     */
+    private HashMap<String, BuildingBuilder.BuildingBuilderResource> resourcesAvailable = new HashMap<>();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Private field 'resourcesAvailable' could be made final; it is only initialized in the declaration or constructor. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AImmutableField)